Src/DeUrgenta.Admin.Api/CommandHandlers/UpdateEventHandler.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using CSharpFunctionalExtensions;
+using DeUrgenta.Admin.Api.Commands;
+using DeUrgenta.Common.Models;
+using DeUrgenta.Common.Validation;
+using DeUrgenta.Domain;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+
+namespace DeUrgenta.Admin.Api.CommandHandlers
+{
+    public class UpdateEventHandler : IRequestHandler<UpdateEvent, Result<EventModel>>
+    {
+        private readonly IValidateRequest<UpdateEvent> _validator;
+        private readonly DeUrgentaContext _context;
+
+        public UpdateEventHandler(IValidateRequest<UpdateEvent> validator, DeUrgentaContext context)
+        {
+            _validator = validator;
+            _context = context;
+        }
+
+        public async Task<Result<EventModel>> Handle(UpdateEvent request, CancellationToken cancellationToken)
+        {
+            var isValid = await _validator.IsValidAsync(request);
+            if (!isValid)
+            {
+                return Result.Failure<EventModel>("Validation failed");
+            }
+            var ev = await _context.Events.FirstAsync(b => b.Id == request.EventId, cancellationToken);
+            ev.Title = request.EventModel.Title ?? ev.Title;
+            ev.Address = request.EventModel.Address ?? ev.Address;
+            ev.Author = request.EventModel.Author ?? ev.Author;
+            ev.City = request.EventModel.City ?? ev.City;
+            ev.ContentBody = request.EventModel.ContentBody ?? ev.ContentBody;
+            if (request.EventModel.EventTypeId != 0) ev.EventTypeId = request.EventModel.EventTypeId;
+            if (request.EventModel.OccursOn != DateTime.MinValue) ev.OccursOn = request.EventModel.OccursOn;
+            ev.OrganizedBy = request.EventModel.OrganizedBy ?? ev.OrganizedBy;
```

Comment:
we should remove ?? and other checks because we should make sure we get here with valid data

Reply:
when you say we should make sure we get valid data, you mean move those checks to validator class or expect frontend to do it?

Reply:
We use FluentValidators for enforcing valid data so here you will get a valid model